### PR TITLE
Fix the disabled test and some minor cleanups

### PR DESCRIPTION
**Pull request by bvahdat**
- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by bvahdat** [comment]
I guess [CAMEL-15747](https://issues.apache.org/jira/browse/CAMEL-15747) can be closed as `BlobContainerOperations#listBlobs` API is used both by the consumer as well as the producer with the corresponding regex support.

The main issue in the tests (including the `@Disabled` one) was that we asserted on different `MockEndpoint`s _after_ sending messages using the corresponding `ProducerTemplate` instead of _before_ which caused race condition.

**Review by oscerd — APPROVED** [review]

File: components/camel-azure-storage-blob/src/test/java/org/apache/camel/component/azure/storage/blob/integration/BlobConsumerITTest.java [hunk 1]
```diff
@@ -128,42 +132,35 @@ void testBatchFilePolling() throws InterruptedException, IOException {
...
     }
 
     @Test
-    @Disabled("This test should be fixed to use mock:resultRegex endpoint instead of mock:resultBatch")
-    void testRegexPolling() throws InterruptedException {
-        Pattern pattern = Pattern.compile(regex);
+    void testRegexPolling() throws Exception {
+        // test regex based
+        final MockEndpoint mockEndpoint = getMockEndpoint("mock:resultRegex");
+        mockEndpoint.expectedMessageCount(15);
```
Comment (DenisIstomin) [inline components/camel-azure-storage-blob/src/test/java/org/apache/camel/component/azure/storage/blob/integration/BlobConsumerITTest.java:163]: I think that if `15` will be increased to `16` (or larger number) test will still pass.

**Review by DenisIstomin — COMMENTED** [review]
Thank you for the improvement. I think that the disabled test is still incorrect. Because endpoint will receive more than 15 messages.

File: components/camel-azure-storage-blob/src/test/java/org/apache/camel/component/azure/storage/blob/integration/BlobConsumerITTest.java (hunk 1 above)
Comment (bvahdat) [reply components/camel-azure-storage-blob/src/test/java/org/apache/camel/component/azure/storage/blob/integration/BlobConsumerITTest.java:162]: Thanks for your feedback. It should be fixed now.

**Review by DenisIstomin — APPROVED** [review]
Awesome, I think that CAMEL-15747 can be closed after a merge.

**Pull request merged**